File: airbyte-integrations/connectors/destination-teradata/src/test/java/io/airbyte/integrations/destination/teradata/util/JSONStructTest.java
package io.airbyte.integrations.destination.teradata.util;

import com.fasterxml.jackson.databind.JsonNode;
import io.airbyte.cdk.db.jdbc.JdbcUtils;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.sql.SQLException;
import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

public class JSONStructTest {
    private JSONStruct struct;
    private final String json = "{\n" +
            "\t\"id\":123,\n" +
            "\t\"name\":\"Pankaj Kumar\",\n" +
            "}";
    @BeforeEach
    void setup() {
        struct = new JSONStruct("JSON",new Object[] {json});
    }

    @Test
    void testGetAttributes() throws SQLException {
        assertEquals(json, struct.getAttributes()[0]);
    }

    @Test
    void testGetAttributesException() {
        SQLException exception = assertThrows(SQLException.class, () -> {
            Map<Integer, String> map = new HashMap<>();
            struct.getAttributes(map);
        });
        String expectedMessage = "getAttributes (Map) NOT SUPPORTED";
        String actualMessage = exception.getMessage();
        assertTrue(actualMessage.contains(expectedMessage));
    }

    @Test
    void testGetSQLTypeName() throws SQLException {
        assertEquals("JSON", struct.getSQLTypeName());
    }
}

File: airbyte-integrations/connectors/destination-teradata/src/test/kotlin/io/airbyte/integrations/destination/teradata/util/JSONStructTest.kt
/*
 * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.teradata.util

import java.sql.SQLException
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

/** Unit tests for the JSONStruct class. */
class JSONStructTest {

    private lateinit var struct: JSONStruct
    private val json =
        """
        {
            "id":123,
            "name":"Pankaj Kumar",
        }
    """.trimIndent()

    /** Setup method to initialize objects before each test. */
    @BeforeEach
    fun setup() {
        struct = JSONStruct("JSON", arrayOf(json))
    }

    /**
     * Test the getAttributes method.
     *
     * @throws SQLException if an SQL exception occurs
     */
    @Test
    @Throws(SQLException::class)
    fun testGetAttributes() {
        assertEquals(json, struct.attributes[0])
    }

    /** Test the getAttributes method when an exception is expected. */
    @Test
    fun testGetAttributesException() {
        val exception =
            assertThrows(SQLException::class.java) {
                val inputMap = mutableMapOf<String, Class<*>>()
                struct.getAttributes(inputMap)
            }
        val expectedMessage = "getAttributes (Map) NOT SUPPORTED"
        val actualMessage = exception.message
        assertTrue(actualMessage!!.contains(expectedMessage))
    }

    /**
     * Test the getSQLTypeName method.
     *
     * @throws SQLException if an SQL exception occurs
     */
    @Test
    @Throws(SQLException::class)
    fun testGetSQLTypeName() {
        assertEquals("JSON", struct.sqlTypeName)
    }
}
